comp/core/flare/builder/builder.go

```diff
@@ -37,6 +37,17 @@ type FlareBuilder interface {
 	// directly from the CLI process and will not contains any runtime informations.
 	IsLocal() bool
 
+	// Log adds a formatted log entry to the flare file associated with this effect.
+	// The log entry is created using the specified format and optional parameters.
+	//
+	// Parameters:
+	//   format: A format string that specifies how the log entry should be formatted.
+	//   params: A variadic parameter list that provides the values to be inserted into the format string.
+	//
+	// Usage example:
+	//   Log("Processing completed in %d ms", elapsedTime)
+	Log(format string, params ...interface{}) error
```

Comment:
Usually log functions which take a format string and then args end with an `f`, and without `f` they just print all their args, so here `Logf` might be better ?
```suggestion
	Logf(format string, params ...interface{}) error
```